skimage/transform/_warps.py

```diff
@@ -106,16 +107,17 @@ def resize(image, output_shape, order=None, mode='reflect', cval=0, clip=True,
                          "dimensions")
 
     if anti_aliasing is None:
-        anti_aliasing = not image.dtype == bool
+        anti_aliasing = not input_type == bool
 
-    if image.dtype == bool and anti_aliasing:
+    if input_type == bool and anti_aliasing:
         warn("Input image dtype is bool. Gaussian convolution is not defined "
              "with bool data type. Please set anti_aliasing to False or "
              "explicitely cast input image to another data type. Starting "
              "from version 0.19 a ValueError will be raised instead of this "
              "warning.", FutureWarning, stacklevel=2)
 
     factors = np.divide(input_shape, output_shape)
+    image = convert_to_float(image, preserve_range)
     # create copy so input values range stays accessible through image for clip
     img_in = image
```

Comment:
```suggestion
    img_in = convert_to_float(image, preserve_range)
```

Reply:
The following `convert_to_float` calls can now be removed :wink:.

Reply:
:pray: You already removed them :+1:

Reply:
Yes, however `convert_to_float` is called in warp, so in the 2D case of the fallback code, it will be called twice... probably better to let it like this than touch also the warp function

Reply:
Arrays pass through `convert_to_float` without modification when their data type is floating point, so there is no major overhead by calling it twice...

Reply:
The proposition you made will not work I think: the reference used for clipping is `image` (as modified in the first commit for this issue). A problem with uint8 data can occur when `preserve_range` is False, so `image` values are in [0, 255] and `img_in` values are in [0, 1].
For having proper reference for clipping, I think `convert_to_float` should be applied to `image`, so references values are in [0,1].

Reply:
:facepalm: you are right :)